src/decompiler-ast.h

```diff
@@ -340,15 +355,21 @@ struct AST {
     if (size != 1) {
       InsertNode(NodeType::Statements, ExprType::Nop, nullptr, size);
     }
+    blocks_closed[cur_block_id] = true;
+    cur_block_id = block_stack.back();
+    block_stack.pop_back();
```

Comment:
Wait, couldn't you just `cur_block_id = block_stack.pop_back();`?

Checking: Apparently not no. Dammit C++

Reply:
Yes, the lack of that function has surprised me many times over.